### DISCUSSION TITLE
Possibility of including Multi Agent Reinforcement Learning

### DISCUSSION THREAD

**Discussion by berttggg** [Toni-SM/skrl#91] [Q&A] [answered]
Hi, I am wondering if recently or lately, will the developers include multi-agent reinforcement learning (MARL) model? If yes, will they be adapted to robotics platforms such as Omniverse Isaac Gym and Issac Sim Orbit? Thank you!

**Comment by Toni-SM** [comment, marked as answer]
Hi @berttggg 

I just published a release (`skrl-v1.0.0-rc.1`) that among its features includes support for multi-agent, both in PyTorch and JAX.
You can find multi-agent examples for Bi-DexHands (a multi-agent benchmark on top of Isaac Gym preview 4).

By default, Omniverse Isaac Gym and Issac Orbit don't include multi-agent examples. However, it is possible, with some changes to the environment API, to implement multi-agent tasks.

**Reply by berttggg** [reply]
Wow. Awesome! 
May I know if the developers will be featuring more MARL models in the near future?

**Reply by Toni-SM** [reply]
Hi @berttggg 

The idea is to gradually add more algorithms... in my free time 👀
Do you have a specific MARL algorithm in mind?

**Reply by berttggg** [reply]
Hi~~
Now most of the MARL algorithm restricts the communication between agents during execution. But in complex tasks such as robotics, they actually break the rule, allowing the agents to communicate during both training and execution to complete tasks. So, maybe an API to ease the process of engineering communication messages between agents will be very helpful?
